parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -827,6 +827,11 @@ public DataWriteBuilder createWriterFunc(
       return this;
     }
 
+    public DataWriteBuilder variantShreddingFunc(VariantShreddingFunction func) {
+      appenderBuilder.variantShreddingFunc(func);
```

Comment:
Should we add null check? something like 
Preconditions.checkArgument(func != null, "Variant shredding function cannot be null");